packages/runtime-core/src/apiLifecycle.ts

```diff
@@ -65,10 +65,11 @@ export function injectHook(
 
 export const createHook =
   <T extends Function = () => any>(lifecycle: LifecycleHooks) =>
-  (hook: T, target: ComponentInternalInstance | null = currentInstance) =>
+  (hook: T, target: ComponentInternalInstance | null = currentInstance) => {
     // post-create lifecycle registrations are noops during SSR (except for serverPrefetch)
-    (!isInSSRComponentSetup || lifecycle === LifecycleHooks.SERVER_PREFETCH) &&
-    injectHook(lifecycle, (...args: unknown[]) => hook(...args), target)
+    ;(!isInSSRComponentSetup || lifecycle === LifecycleHooks.SERVER_PREFETCH) &&
+      injectHook(lifecycle, (...args: unknown[]) => hook(...args), target)
```

Comment:
Would it maybe be clearer to write this using an `if`, rather than using `&&`?

The previous implementation was trying to keep the whole thing as a single expression, but now the arrow function has braces that's no longer required. I can see how writing things like `fn && fn()` can be more convenient than using `if`, but once the condition gets a bit more complicated, like in this code, I think using an explicit `if` makes it much easier for a human to parse what the code is doing.

Reply:
Thanks you!